modules/app/app-contentstudio/src/main/js/app/wizard/page/contextwindow/inspect/page/PageInspectionPanel.ts

```diff
@@ -111,6 +144,27 @@ export class PageInspectionPanel extends BaseInspectionPanel {
         }
     }
 
+    private updateSaveAsTemplateVisibility() {
+        const content = this.liveEditModel.getContent();
+        const hasController = this.pageTemplateSelector.getValue();
+        if (hasController && content.isSite()) {
+            if (this.currentUserHasCreateRights === null) {
+                new api.content.resource.GetPermittedActionsRequest()
+                    .addContentIds(content.getContentId())
+                    .addPermissionsToBeChecked(Permission.CREATE)
+                    .sendAndParse().then((allowedPermissions: Permission[]) => {
+
+                    this.currentUserHasCreateRights = allowedPermissions.indexOf(Permission.CREATE) > -1;
+                    this.saveAsTemplateButton.setVisible(this.currentUserHasCreateRights.valueOf());
+                });
+            } else {
+                this.saveAsTemplateButton.setVisible(this.currentUserHasCreateRights.valueOf());
+            }
+        } else {
+            this.saveAsTemplateButton.setVisible(false);
+        }
+    }
+
     private doSelectTemplate(selectedOption: PageTemplateOption) {
```

Comment:
Code in this method is almost identical to the one changing visibility of the action in PageComponentView's context menu (method with the same name). If we switch this to the SaveAsTemplate action then this code should also be centralized inside the SaveAsTemplateAction class and the button will hide/show based on the action.